modules/core/shared/src/main/scala/gem/math/ObservingNight.scala

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.math
+
+import gem.enum.Site
+
+import cats._
+import cats.data.Validated
+import cats.effect.IO
+import cats.implicits._
+
+import java.time._
+import java.time.format.DateTimeFormatter
+
+import scala.math.Ordering.Implicits._
+
+
+/** An observing night is defined as the period of time from 14:00 on one day
+  * until 14:00 on the following day.  In a timezone that honors daylight saving,
+  * it is sometimes onger and sometimes shorter than a period of 24 hours.  This
+  * is also true of days which contain leap seconds.
+  */
+final case class ObservingNight private (start: Instant, end: Instant, site: Site) extends Night {
+
+  // Sanity check ... should be correct via the companion constructor
+  assert(start < end)
+
+  def previous: ObservingNight =
+    ObservingNight.forInstant(start.minusNanos(1L), site)
+
+  def next: ObservingNight =
+    ObservingNight.forInstant(end, site)
+
+  def format: String =
+    ObservingNight.Formatter.withZone(site.timezone).format(end)
+
+}
+
+object ObservingNight {
+
+  /** The hour, in the local time zone, at which the night is considered to
+    * officially start.
+    *
+    * @group Constants
+    */
+  val LocalNightStartHour: Int =
+    14
+
+  /** The local time at which the night is considered to officially start.
+    *
+    * @group Constants
+    */
+  val LocalNightStart: LocalTime =
+    LocalTime.of(LocalNightStartHour, 0)
+
+  /** Formatter for nights.  The night string representation corresponds to the
+    * date YYYYMMDD for which the night ends in UTC.
+    *
+    * @group Constants
+    */
+  val Formatter: DateTimeFormatter =
+    DateTimeFormatter.ofPattern("yyyyMMdd")
+
+  /** Returns a IO that when executed the observing night corresponding to the
+    * current instant.
+    *
+    * @group Constructors
+    */
+  def now(s: Site): IO[ObservingNight] =
+    IO(Instant.now).map(forInstant(_, s))
+
+  /** Constructs the observing night that includes the given time at the
+    * specified site.
+    *
+    * @group Constructors
+    */
+  def forInstant(i: Instant, s: Site): ObservingNight = {
+    val zdt   = ZonedDateTime.ofInstant(i, s.timezone)
+
+    val twoPM = zdt.withHour(LocalNightStartHour)
```

Comment:
Although 2:00p is probably set in stone the identifier `twoPM` makes me a little uneasy. Could we call this `zonedStart` or something?